File: WebApi.Common/Features/Games/Models/RoundResolvedNotification.cs
namespace WebApi.Common.Features.Games.Models;

public record RoundResolvedNotification(
    string ClaimingPlayerId,
    HandType ClaimedHand,
    List<Rank> Ranks,
    string CallingPlayerId,
    bool WasTruthful,
    string LosingPlayerId,
    int LosingPlayerNewCardCount,
    IReadOnlyDictionary<string, List<Card>> AllPlayerCards,
    string? EliminatedPlayerId,
    string? WinnerPlayerId,
    string? WinnerUsername
);

File: UI/src/features/game/models/RoundResolvedNotification.ts
import type { Card } from "./Card";
import type { HandType } from "./HandType";
import type { Rank } from "./Rank";
import type { Suit } from "./Suit";

export interface RoundResolvedNotification {
    claimingPlayerId: string;
    claimedHand: HandType;
    ranks: Rank[];
    suit: Suit | null;
    callingPlayerId: string;
    wasTruthful: boolean;
    losingPlayerId: string;
    losingPlayerNewCardCount: number;
    allPlayerCards: { [key: string]: Card[] };
    eliminatedPlayerId: string | null;
    winnerPlayerId: string | null;
    winnerUsername: string | null;
}
